calendar/experiments/calendar/parent/ext-calendar-items.js

```diff
@@ -119,6 +120,36 @@ this.calendar_items = class extends ExtensionAPI {
             }
             await pcal.deleteItem(item);
           },
+          query: async function(queryProps) {
+            let calendar = getResolvedCalendarById(context.extension, queryProps.calendarId);
+            let pcal = cal.async.promisifyCalendar(calendar);
+
+            let dateStart = createDate(0, 0, 0);
+            let dateEnd = createDate(9999, 30, 12);
+
+            let filter = queryProps.type ?
+              queryProps.type.reduce((filter, type) => {
+                let thisFilter = 0;
+                switch (type) {
+                  case "completed": thisFilter = Ci.callICalendar.ITEM_FILTER_COMPLETED_ALL; break;
+                  case "classOcurrences": thisFilter = Ci.callICalendar.ITEM_FILTER_CLASS_OCURRENCES; break;
+                }
+
+                return filter | thisFilter;
+              }, 0) :
+              Ci.calICalendar.ITEM_FILTER_ALL_ITEMS;
+
+            // Filter, ???, Date Range Begin, Date Range End
+            return (await pcal.getItems(filter, 0, dateStart, dateEnd))
```

Comment:
FYI I think it's legal to pass `null` for the start and end date rather than making up fictitious dates.